src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -36,8 +36,17 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
             prom_services.append(build_url(scheme='http', host=addr, port=port))
 
             deps.append(dd.name())
+
+        daemons = self.mgr.cache.get_daemons_by_service('mgr')
+        loki_host = ''
+        assert daemons is not None
+        if daemons != []:
+            assert daemons[0].hostname is not None
+            addr = daemons[0].ip if daemons[0].ip else self._inventory_get_addr(daemons[0].hostname)
+            loki_host = build_url(scheme='http', host=addr, port=3100)
```

Comment:
Nowadays you can't go with plain HTTP anymore. We already have some services that still provide plain HTTP, but we'll need to change them soon. For new services, we need to ensure that they are configured in HTTPS. Otherwise this means that the information contained in the logs travels over the network in the clear.

This should be fixed both for Loki and Promtail (assuming none of them is using encryption).

Reply:
@adk3798 , Do you have any idea how to do this?

Reply:
Related: https://github.com/grafana/loki/issues/475